nemo/collections/asr/data/data_simulation.py

```diff
@@ -0,0 +1,1324 @@
+                shutil.rmtree(output_dir)
+                os.mkdir(output_dir)
+            else:
+                raise Exception("Output directory is nonempty and overwrite_output = false")
+        elif not os.path.isdir(output_dir):
+            os.mkdir(output_dir)
+
+        # only add root if paths are relative
+        if not os.path.isabs(output_dir):
+            ROOT = os.getcwd()
+            basepath = os.path.join(ROOT, output_dir)
+        else:
+            basepath = output_dir
+
+        wavlist = open(os.path.join(basepath, "synthetic_wav.list"), "w")
+        rttmlist = open(os.path.join(basepath, "synthetic_rttm.list"), "w")
+        jsonlist = open(os.path.join(basepath, "synthetic_json.list"), "w")
+        ctmlist = open(os.path.join(basepath, "synthetic_ctm.list"), "w")
+        textlist = open(os.path.join(basepath, "synthetic_txt.list"), "w")
+
+        for i in range(0, self._params.data_simulator.session_config.num_sessions):
+            self._furthest_sample = [0 for n in range(0, self._params.data_simulator.session_config.num_speakers)]
+            self._missing_overlap = 0
+
+            logging.info(f"Generating Session Number {i}")
+            filename = self._params.data_simulator.outputs.output_filename + f"_{i}"
+            self._generate_session(i, basepath, filename)
+
+            wavlist.write(os.path.join(basepath, filename + '.wav\n'))
+            rttmlist.write(os.path.join(basepath, filename + '.rttm\n'))
+            jsonlist.write(os.path.join(basepath, filename + '.json\n'))
+            ctmlist.write(os.path.join(basepath, filename + '.ctm\n'))
+            textlist.write(os.path.join(basepath, filename + '.txt\n'))
+
+            # throw error if number of speakers is less than requested
+            num_missing = 0
+            for k in range(0, len(self._furthest_sample)):
+                if self._furthest_sample[k] == 0:
+                    num_missing += 1
+            if num_missing != 0:
+                warnings.warn(
+                    f"{self._params.data_simulator.session_config.num_speakers-num_missing} speakers were included in the clip instead of the requested amount of {self._params.data_simulator.session_config.num_speakers}"
+                )
+
+        wavlist.close()
+        rttmlist.close()
+        jsonlist.close()
+        ctmlist.close()
+        textlist.close()
+
+
+class RIRMultiSpeakerSimulator(MultiSpeakerSimulator):
+    """
+    RIR Augmented Multispeaker Audio Session Simulator.
+    """
+
+    def __init__(self, cfg):
+        super().__init__(cfg)
+        self._check_args_rir()
+
+    def _check_args_rir(self):
+        """
+        Checks YAML arguments to ensure they are within valid ranges
+        """
+
+        if (
+            self._params.data_simulator.rir_generation.toolkit != 'pyroomacoustics'
+            and self._params.data_simulator.rir_generation.toolkit != 'gpuRIR'
+        ):
```

Comment:
Simplify by using: `if not (self._params.data_simulator.rir_generation.toolkit in [''pyroomacoustics", "gpuRIR"])`